Integration Test Scenarios on top of Jörmungandr (Initial Setup)

KtorZ opened the pull request:
# Issue Number

<!-- Put here a reference to the issue this PR relates to and which requirements it tackles -->

#358 

# Overview

<!-- Detail in a few bullet points the work accomplished in this PR -->

- [x] I have updated our `genesis.yaml` (and re-generated `block0.bin`) to comply with newer versions of Jörmungandr.
- [x] I have adjusted the `LauncherSpec` in Jörmungandr to support running the integration scenarios.
- [x] I have initialized and started a wallet server using Jörmungandr
- [x] I have added a dummy integration spec which restores (successfully!) a fixture wallet.

# Comments

<!-- Additional comments or screenshots to attach if any -->

This is a first draft that spins up a Jormungandr node, and starts a wallet server to talk to it directly.
The setup is definitely simpler than for cardano-sl, and also much faster!

One thing I couldn't do was to generate many UTxOs per wallet. The genesis file in newer versions of
Jormungandr doesn't support having duplicate initial funds. A possible fix would be to generate use the
first 10 addresses of each wallet, instead of re-using the same address.

```
Jörmungandr Temporary Spec
(Status {statusCode = 200, statusMessage = "OK"},Right (ApiWallet {id = ApiT {getApiT = WalletId {getWalletId = 37403204154ae83f78b5141f4089fa197424b189}}, addressPoolGap = ApiT {getApiT = AddressPoolGap {getAddressPoolGap = 20}}, balance = ApiT {getApiT = WalletBalance {available = Quantity 100000000000, total = Quantity 100000000000}}, delegation = ApiT {getApiT = NotDelegating}, name = ApiT {getApiT = WalletName {getWalletName = "Faucet Wallet"}}, passphrase = Just (ApiT {getApiT = WalletPassphraseInfo {lastUpdatedAt = 2019-06-22 15:37:45.169367354 UTC}}), state = ApiT {getApiT = Restoring (Quantity (Percentage {getPercentage = 0}))}}))
  Temporary example spec for Jörmungandr integration
```

:tada: 

<!-- 
Don't forget to:

 ✓ Self-review your changes to make sure nothing unexpected slipped through
 ✓ Assign yourself to the PR
 ✓ Assign one or several reviewer(s)
 ✓ Once created, link this PR to its corresponding ticket
 ✓ Acknowledge any changes required to the Wiki
-->

KtorZ committed:
update jormungandr genesis data using jcli latest version

KtorZ committed:
adapt Jormungandr LauncherSpec for integration tests scenarios

This is a first draft that spins up a Jormungandr node, and starts a wallet server to talk to it directly.
The setup is definitely simpler than for cardano-sl, and also much faster!

One thing I couldn't do was to generate many UTxOs per wallet. The genesis file in newer versions of
Jormungandr doesn't support having duplicate initial funds. A possible fix would be to generate use the
first 10 addresses of each wallet, instead of re-using the same address.

KtorZ commented on lib/jormungandr/test/data/jormungandr/genesis.yaml:
```diff
@@ -46,240 +44,51 @@ blockchain_configuration:
-# legacy cardano address format
-legacy_funds:
-  - address: 48mDfYyQn21iyEPzCfkATEHTwZBcZJqXhRJezmswfvc6Ne89u1axXsiazmgd7SwT8VbafbVnCvyXhBSMhSkPiCezMkqHC4dmxRahRC86SknFu6JF6hwSg8
-    value: 123
```
See comment on the PR.

paweljakubas reviewed (approved):
very nice step towards full integration tests with Jormungandr! I see some failures at CI....

KtorZ committed:
bump Jormungandr version to 0.2.3

KtorZ committed:
move 'Cardano.Wallet.Unsafe' into 'core' library

And also move a couple of other 'unsafe' functions there, like
'unsafeRunExceptT' from 'Cardano.Wallet'.
